### PR TITLE
op-deployer: Fee Recipients and Gas Params added to intent

### PR DESCRIPTION
**Pull request by blmalone**
Issue originating from: https://github.com/ethereum-optimism/platforms-team/issues/284 

### Fee Recipients

The existing integration test: `op-chain-ops/deployer/integration_test/apply_test.go` applies a pipeline of tasks for op-deployer. As we know, the last task in the pipeline after performing an `apply` is `generate-l2-genesis` (see [here](https://github.com/ethereum-optimism/optimism/blob/develop/op-chain-ops/deployer/apply.go#L172)).

In this PR, I'm adding configurable fee recipient addresses to the intent file. These addresses will be injected as immutables into three predeploy contracts:

- BaseFeeVault
- L1FeeVault
- SequencerFeeVault

In the `apply_test.go` integration test, I'm verifying that these addresses exist in the runtime bytecode of the three contracts mentioned above. This existence check is a best effort check to ensure that the immutables were injected correctly into the contracts. 

### Gas Params 
Params defined in this PR have been taken from [here](https://github.com/ethereum-optimism/optimism/blob/develop/packages/contracts-bedrock/deploy-config/mainnet.json#L39-L40).

### CODE REVIEW

**Comment by blmalone** [comment]
* **#12404** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/12404?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `develop`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @blmalone and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: op-chain-ops/deployer/init.go
Comment (blmalone) [inline op-chain-ops/deployer/init.go]: default values taken from: https://github.com/ethereum-optimism/optimism/blob/develop/packages/contracts-bedrock/deploy-config/mainnet.json#L39-L40

File: op-chain-ops/deployer/integration_test/apply_test.go
Comment (blmalone) [inline op-chain-ops/deployer/integration_test/apply_test.go:298]: Not sure of a better way to prove the immutable was set other than checking for the existence of it in the implementation contract. 
It's not the strictest test but it gives us confidence that the value in the intent file _was used_ in the predeploy.

If I could execute `.Code` on an EVM and call the `recipient()` function, this would be a stronger test.

File: op-chain-ops/deployer/init.go
Comment (blmalone) [inline op-chain-ops/deployer/init.go]: Defaulting to zero address for fee recipients.

File: op-chain-ops/deployer/integration_test/apply_test.go
Comment (mds1) [reply op-chain-ops/deployer/integration_test/apply_test.go:298]: Agreed that would be a stronger test, but I'm unsure how to do that, and this is probably sufficient because the odds the bytecode contains the same 20 byte string as the specified address is very small

File: op-chain-ops/deployer/init.go
Comment (mds1) [inline op-chain-ops/deployer/init.go]: Looks like the new values here are in the root config, like this:
```toml
l1ChainID = 11155111
fundDevAccounts = true
contractArtifactsURL = "https://storage.googleapis.com/oplabs-contract-artifacts/artifacts-v1-ee07c78c3d8d4cd8f7a933c050f5afeebaa281b57b226cc6f092b19de2a8d61f.tar.gz"
contractsRelease = "op-contracts/v1.6.0"
baseFeeVaultRecipient = "0x0000000000000000000000000000000000000000"
l1FeeVaultRecipient = "0x0000000000000000000000000000000000000000"
sequencerFeeVaultRecipient = "0x0000000000000000000000000000000000000000"
eip1559Denominator = 50
eip1559Elasticity = 6

[[chains]]
  id = "0x0000000000000000000000000000000000000000000000000000000000000002"
  [chains.roles]
    proxyAdminOwner = "0xc2681254d6b08b1eebfd841d75bdaf581c14f810"
    systemConfigOwner = "0x3e4f624ddf5572b89b051f6773ea044582d2f207"
    governanceTokenOwner = "0xc2681254d6b08b1eebfd841d75bdaf581c14f810"
    unsafeBlockSigner = "0x39db188b2058d54016a0ee49dcc59781b5cd67d8"
    batcher = "0x8d394e2ddb85074f395713b613a4d896af3aa898"
    proposer = "0xed5f4989bab532801afff8d629eedd32326ef91d"
    challenger = "0x4ffa5cb5b17e6587821ccf01491bfabc3376675a"
```

but we should make them chain specific, like this, because each chain may use different values for these
```toml
l1ChainID = 11155111
fundDevAccounts = true
contractArtifactsURL = "https://storage.googleapis.com/oplabs-contract-artifacts/artifacts-v1-ee07c78c3d8d4cd8f7a933c050f5afeebaa281b57b226cc6f092b19de2a8d61f.tar.gz"
contractsRelease = "op-contracts/v1.6.0"

[[chains]]
  id = "0x0000000000000000000000000000000000000000000000000000000000000002"
  baseFeeVaultRecipient = "0x0000000000000000000000000000000000000000"
  l1FeeVaultRecipient = "0x0000000000000000000000000000000000000000"
  sequencerFeeVaultRecipient = "0x0000000000000000000000000000000000000000"
  eip1559Denominator = 50
  eip1559Elasticity = 6
  [chains.roles]
    proxyAdminOwner = "0xc2681254d6b08b1eebfd841d75bdaf581c14f810"
    systemConfigOwner = "0x3e4f624ddf5572b89b051f6773ea044582d2f207"
    governanceTokenOwner = "0xc2681254d6b08b1eebfd841d75bdaf581c14f810"
    unsafeBlockSigner = "0x39db188b2058d54016a0ee49dcc59781b5cd67d8"
    batcher = "0x8d394e2ddb85074f395713b613a4d896af3aa898"
    proposer = "0xed5f4989bab532801afff8d629eedd32326ef91d"
    challenger = "0x4ffa5cb5b17e6587821ccf01491bfabc3376675a"
```

File: op-chain-ops/deployer/state/intent.go
Comment (mds1) [inline op-chain-ops/deployer/state/intent.go]: Related to above comment, these would get moved into the `ChainIntent` struct since these values will vary per chain

Comment (blmalone) [reply op-chain-ops/deployer/state/intent.go]: Changed in: https://github.com/ethereum-optimism/optimism/pull/12404/commits/2d3aa8a87c5131f22242dc85bd8209a4ba47a81d

File: op-chain-ops/deployer/init.go
Comment (blmalone) [reply op-chain-ops/deployer/init.go]: Changed in: https://github.com/ethereum-optimism/optimism/pull/12404/commits/2d3aa8a87c5131f22242dc85bd8209a4ba47a81d

**Review by mds1 — APPROVED** [review]
LGTM, but let's get a review from @bitwiseguy before merging

File: op-chain-ops/deployer/init.go
Comment (bitwiseguy) [inline op-chain-ops/deployer/init.go:111]: These will automatically default to their zero value if not specified. So nothing wrong with explicitly setting the values here, but the effect is the same as if they are not specified

File: op-chain-ops/deployer/init.go
Comment (bitwiseguy) [reply op-chain-ops/deployer/init.go]: It would be nice to only hardcode these default values in only one location, then import them in other spots

File: op-chain-ops/deployer/init.go
Comment (blmalone) [reply op-chain-ops/deployer/init.go:111]: Okay, having them as zero for now is okay. I'll keep the `common.Address{}` in unless there's a reason I shouldn't.

**Review by bitwiseguy — APPROVED** [review]

**Pull request merged**
